src/Eloquent/HasDocumentVersion.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MongoDB\Laravel\Eloquent;
+
+trait HasDocumentVersion
+{
+    /**
+     * Auto call on model instance as booting
+     *
+     * @return void
+     */
+    public static function bootHasDocumentVersion(): void
+    {
+        static::saving(function ($model) {
+          $version = $model->getDocumentVersion();
+
+          if (!$version) {
+              $model->setDocumentVersion(static::getCurrentVersion());
+          }
+        });
+
+        static::retrieved(function ($model) {
+            $model->migrateDocumentVersion($model->getDocumentVersion());
```

Comment:
Skip the call if `$this->getDocumentVersion() >= static::getCurrentVersion()`

Reply:
Of course, I don't know why I didn't think of that.